### ISSUE TITLE
Print in apply, got confused by the output

### ISSUE THREAD

**Issue by slimtom95** [jmcarpenter2/swifter#19]
hi guys,

As the sample code and output below, I was confused why a 'id' series was printed (even weird series length) and  0-9 was printed 3 times more.  

The reason using print to file here is lack of memory in my real world task, so I have to write the generated data into disk while apply running.  

Are these redundant outputs inevitable? Am I doing right to use swifter?      

```
# "swifter" == 0.242
import pandas as pd
import swifter
df = pd.DataFrame({'id': [i for i in range(10)]})
f = open('tmpfile', 'w')
_ = df.swifter.apply(lambda x: print(x['id'], file=f))
```
and the content in file is  
```
0    0
1    1
2    2
3    3
4    4
5    5
6    6
7    7
Name: id, dtype: int64
0
1
2
3
4
5
6
7
8
9
0
1
2
3
4
5
6
7
8
9
0
1
2
3
4
5
6
7
8
9
0
1
2
3
4
5
6
7
8
9
```
and of course expected output (by normal pandas apply)
```
0
1
2
3
4
5
6
7
8
9
```

**Comment by jmcarpenter2** [comment]
Hi @slimtom95, 

Sorry for the delay in my response. I was pretty confused by what you meant. 
After looking into it further, this issue is fundamental to the way that swifter works. Let me explain.

First, pandas tries to vectorize, returning the first part of your output:

> 0    0
> 1    1
> 2    2
> 3    3
> 4    4
> 5    5
> 6    6
> 7    7
> Name: id, dtype: int64

Since this failed, next it does 3 timing tests on how long the first up to 1000 rows of the apply will take

> 0
> 1
> 2
> 3
> 4
> 5
> 6
> 7
> 8
> 9
> 0
> 1
> 2
> 3
> 4
> 5
> 6
> 7
> 8
> 9
> 0
> 1
> 2
> 3
> 4
> 5
> 6
> 7
> 8
> 9

Finally, it decides that pandas is the quickest method of performing this apply, so it returns the expected pandas output.

> 0
> 1
> 2
> 3
> 4
> 5
> 6
> 7
> 8
> 9

Long story short, I don't think using swifter to print is a smart idea. If you could elaborate on the purpose of swifter in this use case then I can potentially look into expanding this issue into a broader one and try to derive a better response/solution for you. I do see that there is an argument here for suppressing print statements while doing the 3 timing calls though.

**Comment by slimtom95** [comment]
Thanks for your detailed answer. It's a shame that I didn't look into the source code and I find the code is quite clear to explain this.

The purpose using swifter here is try to gain some performance benefits of parallel computing since the process function works on a series. And the reason of using print to write data into disk is the memory of my pc cannot fit the generated data while swifter.apply running.

I don't find a way to suppress the output of applied func in testing phase. It seems that the applied func should only concentrate on the computation of target data and any io or changing out scope data would be risky and not appropriate.

I'm closing the issue, jmcarpenter2,  thank you
